zerocopy-derive/Cargo.toml

```diff
@@ -28,7 +28,7 @@ proc-macro = true
 [dependencies]
 proc-macro2 = "1.0.1"
 quote = "1.0.10"
-syn = "2.0.46"
+syn = { version = "2.0.46", features = ["full"] }
```

Comment:
@djkoloski Do you have a sense of how we could implement using the same trick that you pulled to avoid the `full` feature?

I tested locally on a MacBook Pro on low battery mode, and found that the `full` feature increased `zerocopy-derive` `cargo build` time from 8.9s to 13.3s. That's a large percentage, but a small absolute difference. So I'm not opposed to just sticking with this impl, but it'd be cool if we could avoid it.